src/unistd.rs

```diff
@@ -1326,6 +1327,45 @@ pub fn pause() -> Result<()> {
     Errno::result(res).map(drop)
 }
 
+/// Schedule an alarm signal.
+///
+/// Returns a previously set alarm, if any.
+///
+/// See also [alarm](http://pubs.opengroup.org/onlinepubs/9699919799/functions/alarm.html).
+///
+/// # Note
+///
+/// Alarm only allows for times with a second granularity, anything smaller then
+/// a second will be ignored (not rounded up). E.g. 1 second and 800 miliseconds
+/// will become 1 second.
+///
+/// Also, Strictly conforming the POSIX spec, duriations longer then `65535`
+/// seconds may overflow the type used in the raw `alarm` function.
+pub fn alarm(from_now: Duration) -> Option<Duration> {
+    let secs = from_now.as_secs();
+    let old_secs = unsafe { libc::alarm(secs as libc::c_uint) };
+    if old_secs != 0 {
+        Some(Duration::from_secs(old_secs as u64))
+    } else {
+        None
+    }
+}
+
+/// Cancels an alarm signal.
+///
+/// The alarm signal must be previously set by [`alarm`], and will be returned
+/// if any is canceld.
+///
+/// [`alarm`]: fn.alarm.html
+pub fn cancel_alarm() -> Option<Duration> {
```

Comment:
Is this function really necessary?  Can't we accomplish the same thing just by making `alarm`'s argument an `Option<Duration>`?

Reply:
I will change it.